crypto/evp/kem.c

```diff
@@ -142,7 +143,10 @@ static int evp_kem_init(EVP_PKEY_CTX *ctx, int operation,
     }
 
     if (provkey == NULL) {
-        EVP_KEM_free(kem);
+        if (kem != NULL) {
```

Comment:
`EVP_KEM_free(NULL)` is a no-op, so it's safe to call on `NULL`, so this test isn't needed

Reply:
Committed the changes again after removing NULL test while keeping the NULL assignment intact.